uiowa/uiowa
# Person Contact Information

Resolves #2297 

## Does

- Provides paragraphs "field collection" allowing multiple contact entries, split by optional label.
- Address field with all fields optional
  - Address has first name, last name, company as well as address line 2 to handle building/room/etc.
- Phone field (optional)
- Available on hr.uiowa.edu with a bare minimum implementation.
- Gracefully imports to person_extended websites. Since the form/displays config is split, these fields exist but are not available for use.
- Despite unusual markup structure, there are no a11y flags from SiteImprove related to this new functionality.

## Does Not

- BUILDING: VARCHAR2(4000) NULL
- ROOM: VARCHAR2(4000) NULL
- STAFF_CONTACT: VARCHAR2(128) NULL
- Separate fields for fax, cell, etc.

## To Test

- pull, `composer install && blt frontend && blt dsa`
- As a webmaster you should be able to add separate entries of contact information on v2, v3, HR and collegiate sites without a separate split for person content type form/display (e.g. person_extended)

## Notes

- Frontend blank space is registered if node is saved with a blank contact item as it technically exists as a field item of the parent paragraphs field. Logic could be added to account for this, but so far the space isn't really affecting anything.

![default local drupal uiowa edu_people_herky-hawk (2)](https://user-images.githubusercontent.com/4663676/97755974-1ef3af00-1ac8-11eb-88ba-c0b3222dfb9a.png)
